rocprim/include/rocprim/block/detail/block_reduce_warp_reduce.hpp

```diff
@@ -178,21 +178,25 @@ class block_reduce_warp_reduce
             input, output, num_valid, reduce_op
         );
 
-        // i-th warp will have its partial stored in storage_.warp_partials[i-1]
-        if(lane_id == 0)
+        // Final reduction across warps is only required if there is more than 1 warp
+        if (warps_no_ > 1)
```

Comment:
@ex-rzr was working on a similar fix internally. From what I can see the only difference was the use of `ROCPRIM_IF_CONSTEXPR`.

```suggestion
        if ROCPRIM_IF_CONSTEXPR (warps_no_ > 1)
```

Reply:
Ah right, that would be better. Implemented, thanks!